packages/connect-popup/e2e/tests/permissions.test.ts

```diff
@@ -18,23 +19,49 @@ const fixtures = [
     {
         description: `iframe and host same origins but with trust-issues=true -> show permissions`,
         queryString: '?trust-issues=true',
+        setTrustedHost: false,
         expect: () => popup.waitForSelector("[data-test='@permissions/confirm-button']"),
     },
     {
         description: `iframe and host same origins but with trust-issues=false -> no permissions shown`,
         queryString: '?trust-issues=false',
+        setTrustedHost: false,
         expect: () =>
             popup.waitForSelector('//p[contains(., "Follow instructions on your device")]', {
                 state: 'visible',
                 strict: false,
             }),
     },
+    {
+        description:
+            'iframe and host same origins but with settings to trustedHost -> no permissions shown',
+        queryString: '',
+        setTrustedHost: true,
+        trustedHost: true,
+        expect: () =>
+            popup.waitForSelector('//p[contains(., "Follow instructions on your device")]', {
+                state: 'visible',
+                strict: false,
+            }),
+    },
+    {
+        description:
+            'iframe and host same origins but with settings to NOT trustedHost -> show permissions',
+        queryString: '',
+        setTrustedHost: false,
+        expect: () => popup.waitForSelector("[data-test='@permissions/confirm-button']"),
+    },
 ];
 
 fixtures.forEach(f => {
     test(f.description, async () => {
         const browserInstance = await chromium.launch();
         const page = await browserInstance.newPage();
+        if (f.setTrustedHost) {
+            await page.goto(`${url}#/settings`);
+            await waitAndClick(page, ['@checkbox/trustedHost']);
+            await waitAndClick(page, ['@submit-button']);
+        }
         await page.goto(`${url}${f.queryString}#/method/verifyMessage`);
```

Comment:
page.goto does not cause page to refresh unloading all scripts? just asking..

Reply:
so this is not reload, only navigation I guess. triggering actions bases on route changes from some middleware, right?

Reply:
Yes, that is right.